### ISSUE TITLE
TypeError: e.getIdToken is not a function

### ISSUE THREAD

**Issue by louisnt-prod** [gladly-team/next-firebase-auth#128]
**Describe the bug**
Type error when launching next-firebase-auth linked to getIdToken function.

**Version**
 ```
   "firebase": "^8.3.2",
    "firebase-admin": "^9.6.0",
    "next": "^10.1.2",
    "next-firebase-auth": "^0.13.0-alpha.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
```

**To Reproduce**

1. Wrap a static landing page with withAuthUser
`export default withAuthUser()(LandingPage);
`
2. Launch
`yarn dev`

3. The error is linked to createAuthUser

```
Uncaught (in promise) TypeError: e.getIdToken is not a function
    at Object.eval (index.browser.js?3144:1)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js?c973:3)
    at _next (asyncToGenerator.js?c973:25)
    at eval (asyncToGenerator.js?c973:32)
    at new Promise (<anonymous>)
    at Object.eval (asyncToGenerator.js?c973:21)
    at Object.eval [as getIdToken] (index.browser.js?3144:1)
    at eval (index.browser.js?3144:1)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js?c973:3)
    at _next (asyncToGenerator.js?c973:25)
    at eval (asyncToGenerator.js?c973:32)
    at new Promise (<anonymous>)
    at eval (asyncToGenerator.js?c973:21)
    at eval (index.browser.js?3144:1)
    at eval (index.browser.js?3144:1)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js?c973:3)
    at _next (asyncToGenerator.js?c973:25)
    at eval (asyncToGenerator.js?c973:32)
    at new Promise (<anonymous>)
    at eval (asyncToGenerator.js?c973:21)
    at eval (index.browser.js?3144:1)
    at Object.eval (index.browser.js?3144:1)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js?c973:3)
    at _next (asyncToGenerator.js?c973:25)
```

**Expected behavior**
Understand where the error comes from and correct it

**Additional context**

Thank you for your help!

**Comment by louisnt-prod** [comment]
FYI, I downgraded to 0.12.2, no issue with this version if someone is facing the same issue.

**Comment by Twister42** [comment]
Same problem here, works with alpha.0

**Comment by aprendendofelipe** [comment]
This problem starts in 0.13.0-alpha.2

**Comment by kmjennison** [comment]
Thanks for the report. I'm able to reproduce and am working on a fix. It looks like the problem is that we used the spread operator to modify the JS SDK Firebase user (in `useFirebaseUser.js`), but doing that drops its methods.

**Comment by litewarp** [comment]
@kmjennison Just about to say that!

The easiest fix is to change the function in createAuthUser to  
```
getIdToken: async () => firebase.auth.currentUser.getIdToken()
```

But that blows up the test coverage, but it feels like you're better at that, so hopefully it won't take too long. The other option is to pass the user and the claims separately to the `setAuthCookie` function.

**Comment by kmjennison** [comment]
Fixed in [v0.13.0-alpha.4](https://github.com/gladly-team/next-firebase-auth/releases/tag/v0.13.0-alpha.4) and updated the live example app.
